pkg/metrics/delivery_reporter.go

```diff
@@ -185,3 +198,11 @@ func getStartDeliveryProcessingTime(ctx context.Context) (time.Time, error) {
 	}
 	return time.Time{}, fmt.Errorf("missing or invalid start time: %v", v)
 }
+
+func getRespStatusCode(ctx context.Context) (int, error) {
```

Comment:
nit: based on the usage, a better signature seems to be `getRespStatusCode(...) (int, bool)`

Reply:
done